### PR TITLE
Task 4 tests

### PR DESCRIPTION
**Pull request by hmaxim**

### CODE REVIEW

File: movie-app/src/shared-components/movie-item/MovieItem.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ const MovieItem = (props: IMovie) => {
   }, []);
 
   return (
-    <MovieItemWrapper onClick={() => navigate()}>
+    <MovieItemWrapper onClick={() => navigate}>
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/movie-item/MovieItem.tsx:12]: You no longer invoke navigate method
+ you can bass `navigate` as is, it's not needed to create a new function

File: movie-app/src/shared-components/movie-item/MovieItemWrapper.tsx [hunk 2]
```diff
@@ -26,4 +26,6 @@ const MovieItemWrapper = styled.div`
   }
 `;
 
+MovieItemWrapper.displayName = "MovieItemWrapper";
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/movie-item/MovieItemWrapper.tsx:29]: Is it needed? you don't seem to use enzyme's `find` method with string argument

File: movie-app/src/shared-components/movie-item/MovieItem.test.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  test("should have props", () => {
+    expect(component.props).toBeTruthy()
+  });
+
+  test("should be rendered", () => {
+    expect(toJson(component.find(MovieItemWrapper))).toMatchSnapshot();
+  });
+
+  test("should have movie-poster", () => {
+    expect(toJson(component.find(".movie-poster"))).toBeDefined();
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/movie-item/MovieItem.test.tsx]: or `expect(component.find('.movie-poster')).toHaveLength(1)` (or whatever number of items you have). At the same time, such tests are redundant as you already have tested the component's view with snapshot

File: movie-app/src/shared-components/movie-item/MovieItem.test.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
+  budget: 0,
+  revenue: 0,
+  runtime: 150,
+  genres: ["Action"]
+};
+
+
+describe("MovieItem Component", () => {
+  const component = mount(<MovieItem {...MOVIE} />);
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/movie-item/MovieItem.test.tsx]: having one wrapper instance for all tests violates 1 of most important unit testing principles: you test should be independent. By sharing single instance one unit test may affect results of another. To fix that you can use `beforeEach` test hook, and create a new mount instance on each test

File: movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    active: false
+  },
+  {
+    label: "test2",
+    active: true
+  }
+];
+
+describe("SwitchersButtons component", () => {
+  const component = mount(<SwitchersButtons searchButtonsParams={buttons} />);
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx]: same, tests should not share a single wrapper

File: movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  test("should have props", () => {
+    expect(component.props).toBeTruthy();
+  });
+
+  test("should define search-buttons-container", () => {
+    expect(toJson(component.find(".search-buttons-container"))).toBeDefined();
+  });
+
+  test("should define search-buttons-container", () => {
+    expect(toJson(component.find(SwitchersButtonWrapper))).toHaveLength(2);
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx]: I think toJson is redunant here

File: movie-app/jest.config.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+  //   "js",
+  //   "json",
+  //   "jsx",
+  //   "ts",
+  //   "tsx",
+  //   "node"
+  // ],
+
+  // A map from regular expressions to module names that allow to stub out resources with a single module
+  // moduleNameMapper: {},
```
Comment (AlexeyBarkow) [inline movie-app/jest.config.js]: For now I'm unable to collect coverage for your tests. Please, specify moduleNameMapper for .jpg files

File: movie-app/src/shared-components/movie-item/MovieItem.test.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import "jsdom-global/register";
```
Comment (AlexeyBarkow) [inline movie-app/src/shared-components/movie-item/MovieItem.test.tsx]: generated snapshots should be commited

File: .gitignore [hunk 9]
```diff
@@ -5,6 +5,9 @@ dist/
 ./movie-app/dist
```
Comment (AlexeyBarkow) [inline .gitignore:5]: Unit tests coverage is insufficient. I've run coverage report for all project files and see the following picture:
```
29.87% Statements 23/77
50% Branches 12/24
42.86% Functions 12/28
30.67% Lines 23/75
```

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: movie-app/src/shared-components/movie-item/MovieItem.tsx (hunk 1 above)
Comment (hmaxim) [reply movie-app/src/shared-components/movie-item/MovieItem.tsx:12]: Understood

File: movie-app/src/shared-components/movie-item/MovieItemWrapper.tsx (hunk 2 above)
Comment (hmaxim) [reply movie-app/src/shared-components/movie-item/MovieItemWrapper.tsx:29]: Not needed.

File: movie-app/src/shared-components/movie-item/MovieItem.test.tsx (hunk 3 above)
Comment (hmaxim) [reply movie-app/src/shared-components/movie-item/MovieItem.test.tsx]: Yes, the same percentage of coverage.

File: movie-app/src/shared-components/movie-item/MovieItem.test.tsx (hunk 4 above)
Comment (hmaxim) [reply movie-app/src/shared-components/movie-item/MovieItem.test.tsx]: fixed

File: movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx (hunk 5 above)
Comment (hmaxim) [reply movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx]: fixed

File: movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx (hunk 6 above)
Comment (hmaxim) [reply movie-app/src/shared-components/switchers-buttons/SwithcersButtons.test.tsx]: fixed

File: movie-app/jest.config.js (hunk 7 above)
Comment (hmaxim) [reply movie-app/jest.config.js]: Fixed

**Pull request merged**